framework/src/node/forger/forger.ts

```diff
@@ -191,15 +223,52 @@ export class Forger {
 			);
 		}
 
-		const account = await this._chainModule.dataAccess.getAccountByAddress(forgerAddress);
-
-		if (forging) {
-			this._keypairs.set(forgerAddress, keypair);
-			this._logger.info(`Forging enabled on account: ${account.address.toString('hex')}`);
-		} else {
+		if (!forging) {
+			// Disable delegate by removing keypairs corresponding to address
 			this._keypairs.delete(forgerAddress);
-			this._logger.info(`Forging disabled on account: ${account.address.toString('hex')}`);
+			this._logger.info(`Forging disabled on account: ${forgerAddress.toString('hex')}`);
+			return {
+				address: forgerAddress,
+				forging,
+			};
+		}
+
+		const previouslyForgedMap = await getPreviouslyForgedMap(this._db);
+		const forgingInput = { height, maxHeightPreviouslyForged, maxHeightPrevoted };
+		if (forgerInfoMissing(previouslyForgedMap, forgerAddress, forgingInput, overwrite)) {
```

Comment:
This case and the next if condition all treat the case the we do not overwrite. In this case the previous forger info has to exist and equal the one given as input. I think the code is correct, but it is a bit hard to see that all cases are covered. We also do a few redundant checks. Maybe the structure could be

if not synced:
   throw error
if not overwrite
  check that forger info exists
  check that forger info is equal
else
  set forger info as given in input

Reply:
Maybe the check could also be moved back in place to make it a bit easier to read. They are also not re-used anywhere else.